graphix/pattern.py

```diff
@@ -2179,3 +2180,20 @@ def assert_permutation(original, user):
             node_set.remove(node)
         else:
             assert False, f"{node} appears twice"
+
+
+@dataclass
+class ExtractedSignal:
+    s_domain: list[int]
+    t_domain: list[int]
+    signal: list[int]
```

Comment:
what's the `signal` of `ExtractedSignal`? is this different from domains?

Reply:
`signal` is the domain that the shift command will carry (and `s_domain` and `t_domain` are the new respective domain for the `M` command). I added some doc-comments in 4401e62 to clarify.